tools/gen_nvfuser_version.py

```diff
@@ -11,14 +11,18 @@
 
 # note that this root currently is still part of pytorch.
 def get_sha() -> str:
+    import os
+
+    # assume the $NVFUSER_VERSION is in sha form
+    nvfuser_version = os.environ.get("NVFUSER_VERSION")
     try:
         return (
             subprocess.check_output(["git", "rev-parse", "HEAD"], cwd=nvfuser_root)
             .decode("ascii")
             .strip()
         )
     except Exception:
-        return UNKNOWN
+        return nvfuser_version if nvfuser_version is not None else UNKNOWN
```

Comment:
The priority of sources isn't clear at the first glance. Consider the following: 

```
completed = subprocess.run(..., check=False, capture_output=True)
if completed.returncode == 0:
  return completed.stdout.decode(...)

if nvfuser_version := os.environ.get("NVFUSER_VERSION"):
  assert nvfuser_version is indeed in sha form
  return nvfuser_version

return UNKNOWN
```

Reply:
This way, it's immediately clear to readers that the function prioritizes the git report over the env var.